### PR TITLE
Update Podfile.lock modified timestamp when pod install is run

### LINKED ISSUE #78677: "pod install" is running on every iOS build with plugins, even if nothing changed
```
$ cd dev/integration_tests/flutter_gallery
$ flutter build ios
...
Running pod install...                                           1,605ms
```
Run the same command again, and it will run `pod install` again even though it doesn't need to.  This adds an extra 1.6 seconds to every build.

### PR DESCRIPTION
**Pull request by jmagman**
`pod install` takes ~1.6 seconds on my machine.  The tool tries to only run that command when it's needed:
https://github.com/flutter/flutter/blob/c3d0f60407f4fe556a73f9bef495aac352d4595b/packages/flutter_tools/lib/src/macos/cocoapods.dart#L292-L298

So `2. Podfile.lock is older than Podfile` is problematic when Flutter required `ios/Podfile` and `macos/Podfile` updates like https://github.com/flutter/flutter/pull/59044.  Our own `dev/integration_tests/flutter_gallery` had an older `Podfile.lock` than `Podfile`, so was getting run on every build.

`touch` the Podfile.lock after running `pod install` to make sure it's newer than the `Podfile` to no longer defeat that optimization.

Fixes https://github.com/flutter/flutter/issues/78677

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart [hunk 1]
```diff
@@ -337,11 +337,14 @@ void main() {
     testWithoutContext('exits if Podfile creates the Flutter engine symlink', () async {
```
Comment (xster) [inline packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart:337]: Can we add an catch-all test that breaks without this fix to test that it indeed doesn't rerun pod install if nothing changed?

**Review by xster — APPROVED** [review]

File: packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart]: Added.

File: packages/flutter_tools/test/integration.shard/macos_content_validation_test.dart [hunk 2]
```diff
@@ -118,6 +131,15 @@ void main() {
...
       );
 
+      // Build again without cleaning.
+      final ProcessResult secondBuild = processManager.runSync(buildCommand, workingDirectory: workingDirectory);
+
+      print(secondBuild.stdout);
+      print(secondBuild.stderr);
+      expect(secondBuild.exitCode, 0);
+
+      expect(secondBuild.stdout, isNot(contains('Running pod install')));
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/macos_content_validation_test.dart:141]: Verified this test fails on master:
```
02:35 +0 -2: flutter build macos --release builds a valid app [E]                                                                                                                                                 
  Expected: not contains 'Running pod install'
    Actual: '\n'
              '💪 Building with sound null safety 💪\n'
              '\n'
              'Running pod install...                                           1,443ms\n'
              'Building macOS application...                                   \n'
```

File: dev/integration_tests/flutter_gallery/macos/Podfile.lock [hunk 3]
```diff
@@ -0,0 +1,35 @@
+PODS:
```
Comment (jmagman) [inline dev/integration_tests/flutter_gallery/macos/Podfile.lock:1]: This should be checked in but wasn't.

**Pull request merged**